src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,88 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    public static final String SPLITTER_REGEX = "\\W+";
+    public static final String BUY_STRING_VALUE = "buy";
+    public static final String SUPPLY_STRING_VALUE = "supply";
+    public static final String RESULT_STRING_VALUE = "result";
+    public static final String COMA_STRING = ",";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String data = readFromFile(fromFileName);
+        String report = createReport(data);
+        writeToFile(toFileName, report);
+    }
+
+    private class DataCounter {
+        private int buyAmount = 0;
+        private int supplyAmount = 0;
+
+        private DataCounter() {
+        }
+
+        private void countValues(String[] dataArray) {
+            for (int i = 0; i < dataArray.length; i += 2) {
+                switch (dataArray[i]) {
+                    case BUY_STRING_VALUE -> buyAmount += Integer.parseInt(dataArray[i + 1]);
+                    case SUPPLY_STRING_VALUE -> supplyAmount += Integer.parseInt(dataArray[i + 1]);
+                    default -> throw new RuntimeException(
+                            "Unexpected data in countBuyAndSupplyValues()");
+                }
+            }
+        }
+
+        private int getSupplyAmount() {
+            return supplyAmount;
+        }
+
+        private int getBuyAmount() {
+            return buyAmount;
+        }
+    }
```

Comment:
why do you need this inner class?

Reply:
and inner class should be placed in the end of class

Reply:
When cycle was in the createReport() method, bot did not aprove pull request.  I decided that it was better to make an innner class than to create two methods to count buyAmount and supplyAmount separately using two loops.